### PR TITLE
Fix: Mixing Battle Masters From Different Sub-Factions Does Not Grant Horde Bonus

### PR DESCRIPTION
**Pull request by commy2**
ZH 1.04

- 4 Nuke Battle Masters and 1 Tank Battle Master do not count as a Horde, but 4 Nuke Tank Hunters and 1 Tank Tank Hunter do. 
- The unit type has to match exactly instead of having any of 5 hordable tanks\vehicles.

After ZH 1.04

- Any 5 Battle Masters will form a horde, no matter what sub-faction they are from.

I think this is a thing, because in CCG only one vehicle with horde existed, while there always were two different infantry units.

### CODE REVIEW

File: Patch104pZH/GameFilesEdited/Data/INI/Object/TankGeneral.ini [hunk 1]
```diff
@@ -2798,13 +2798,15 @@ Object Tank_ChinaTankBattleMaster
...
+  ; Patch104p @bugfix commy2 09/09/2021 Fixed horde bonus required exact match when mixing sub-faction Battle Masters.
+
   Behavior = HordeUpdate ModuleTag_04
     RubOffRadius = 150    ; if I am this close to a real hordesman, I will get to be an honorary hordesman
     UpdateRate = 1000     ; how often to recheck horde status (msec)
     Radius = 75           ; how close other units must be to us to count towards our horde-ness (~30 feet or so)
     KindOf = VEHICLE      ; what KindOf's must match to count towards horde-ness
     AlliesOnly = Yes      ; do we only count allies towards horde status?
-    ExactMatch = Yes      ; do we only count units of our exact same type towards horde status? (overrides kindof)
+    ExactMatch = No       ; do we only count units of our exact same type towards horde status? (overrides kindof)
```
Comment (xezon) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/TankGeneral.ini:2809]: How does this know that another faction Battlemaster is ok for horde but a Gattling Tank is not?

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TankGeneral.ini:2809]: None of the Gattling tanks have a HordeUpdate module. They would need a HordeUpdate module, that is also set to `ExactMatch = No`.

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TankGeneral.ini:2809]: For example, 5 Hackers and a Red Guard do not grant Horde for the Red Guard either.

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TankGeneral.ini:2809]: Hmm. And Tank Hunters? Will 2 Battlemaster Horde with 3 Tank Hunters?

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TankGeneral.ini:2809]: Nope. The HordeModule for the Red Guard and Tank Hunter only counts for `KindOf = INFANTRY`.
The HordeModule for the Battle Master(s) however only counts for `KindOf = VEHICLE`.
They do not count cross type.

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TankGeneral.ini:2809]: Ok that sounds very reasonable then.

**Review by xezon — APPROVED** [review]

**Pull request merged**